### ISSUE TITLE
Dependency conflict for Python3.8

### ISSUE THREAD

**Issue by kwat0308** [cescalara/fancy#31]
Currently, using this package with Python <3.9 is not supported due to the dependency of the `ligo.skymap` package. However, using the Galactic Magnetic Field Lensing with CRPropa3 is currently only support up to Python3.8 due to issues with SWIG. While this may be fixed in the future, currently utilising this package with the lensing feature leads to dependency conflicts. The only way to resolve this is to manually comment out the installation dependency in `setup.cfg` and the import in `fancy/plotting/allskymap.py`. 

There should be a simple way to remove the dependency for `ligo.skymap` for python versions < 3.9. This will not affect any important part of the interface as this package is only used for plotting purposes, which can be done through `cartopy` for Py<3.9.

**Comment by cescalara** [comment]
I agree - could we use the `try`/`except` structure + a warning message or so when importing `ligo.skymap` for plots? I think I set up this structure for `crpropa` e.g. [here](https://github.com/cescalara/fancy/blob/97ab041f99192cd35d5690652839f7eba07a27bb/fancy/propagation/gmf.py#L42)

I think the easiest way is to setup the dependencies for the package without conflicts, then document the required setup for e.g. GMF lens. What do you think?

**Comment by kwat0308** [comment]
I think that works. There are anyways two options for skymap plotting, and we dont have to force installation for either one and rather let the user choose. As long as we document this properly I think this will work fine.
